Tiled Gallery: fix view rendering issues

monsieur-z opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/issues/37194

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
This PR fixes the Tiled Gallery not being rendered as expected in the frontend. The issues was introduced by https://github.com/Automattic/jetpack/pull/36854.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
n/a

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No.

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

- Don't checkout this branch just yet
- Spin up a test site
- Connect Jetpack
- Create a new post
- Add a _Tiled Gallery_ block and select some images
- Checkout this branch
- Refresh the editor and verify there's no error message
- In the block styles panel, select _Circles_
<img width="276" alt="Screenshot 2024-05-03 at 10 56 47 AM" src="https://github.com/Automattic/jetpack/assets/1620183/22a71db9-ddd4-4462-a98d-9bd7df8eec90">

- Open the preview
- Notice the items are circular
<img width="500" alt="Screenshot 2024-05-03 at 10 54 27 AM" src="https://github.com/Automattic/jetpack/assets/1620183/e717acfb-6606-4eba-a22d-4032f968971a">

- Verify the other styles are rendered the same in the editor and frontend

Commit:
Tiled Gallery: fix view rendering issues

coder-karen reviewed (commented):
This works well in preview and for published pages. 
However I notice in testing locally and with a Jurassic Ninja site that in the editor the images stay as squares. Checking back a few Jetpack versions I can see the new styles (circles) were previously showing in the editor as well as on the front-end.

jeherve commented:
> In the editor the images stay as squares.

Yes, that's what I'm seeing in my tests as well.

monsieur-z commented:
The gallery was similar to the preview when I initially tested the PR, but testing again it's now different indeed! Not sure what I did there. Thanks for catching this.

Commit:
Update saved block

monsieur-z commented:
I've devised a new solution and updated the block `save` function to move the style-related class name one step higher. A new deprecated version was required for backward compatibility, so I've updated the instructions to take this into account.

Commit:
Ensure gallery is rendered correctly in exising posts

monsieur-z commented on projects/plugins/jetpack/extensions/blocks/tiled-gallery/view.scss:
```diff
@@ -6,12 +6,17 @@ $tiled-gallery-max-column-count: 20;
-	&.is-style-circle .tiled-gallery__item img {
-		border-radius: 50%;
+	&.is-style-circle,
+	.is-style-circle {
```
Ensure the gallery is rendered correctly in exising posts.

coder-karen committed:
Merge branch 'trunk' into fix/tiled-gallery-view

coder-karen reviewed (approved):
Works well for me!

Testing with a previously added gallery, there were no errors when updating the branch either.
